broadinstitute/single_cell_portal_core
# Show cell count beside each filter, update on selection (SCP-5378)

This shows adaptive cell counts beside each filter.  It surfaces category distributions, and how they change upon filtering.

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/5796e260-187b-4688-b4a0-451514707d91

## Overview
### Show cell count beside each filter
Previously (#1904), seeing the distribution of cells across groups in an annotation required changing the "Annotation" (a.k.a. "Color by") menu.  Distributions were only reported in the scatter plot, via the legend.  This made it tedious to assess cell distributions across annotations.

Now, cell counts are reported not only beside each entry in the scatter plot legend, but also beside each filter in all facets.  This makes it easy to determine how cells are distributed across _all_ annotated categories.

### Update on selection
The new filter counts are dynamic.  Each time you check or uncheck a filter, counts for all filters in all facets update to reflect the number of cells matched by the new selection.  These adaptive counts are a powerful way to summarize results of slicing and dicing single-cell datasets across annotations.

## Test
Automated tests were updated to account for these changes.  To manually test:
1.  Go to "Human milk - differential expression" with cell filtering enabled
2. Click "Cell filtering"
3. Note numbers beside each filter
4. In "Cell type" facet, click "More..."
5. Confirm "animal cell" does not appear (it has 0 matching cells in the clustering; its prior display was a bug)
6. Click a filter to deselect it
7. Confirm the count beside the deselected filter is 0
8. Confirm counts beside filters in other facets update

This relates to SCP-5378, a Developer's Choice Days project.
